### DISCUSSION TITLE
uploader 自定义上传，获取到的值是 tempFilePath, 并不是上传后的文件链接。

### DISCUSSION THREAD

**Discussion by GreatAuk** [jdf2e/nutui#2275] [Q&A] [answered]
微信小程序，taro 和 nutui 都是最新版本。
参考官方文档上的 [自定义上传方式(before-xhr-upload)](https://nutui.jd.com/taro/vue/4x/#/zh-CN/component/uploader)
上传成功，但是 `v-model` 绑定的 `fileList` 中的 `url` 还是小程序的 `tempFilePath`。

目前的处理方法是在组件的 `success` 事件中手动更新 `ur`l（其中 `data` 是 `before-xhr-upload` 事件中获取到的文件链接）。
```ts
const beforeXhrUpload = async (_: any, options: any) => {
  try {
    const url = await uploadFileToServer(options.taroFilePath);
    options.onSuccess?.(url, options);
  } catch (err) {
    options.onFailure?.(err, options);
    console.error(err)
  }
}
```
```ts
const onSuccess = ({ data, fileItem }: {data: string, fileItem: FileItem}) => {
  fileItem.name = data
  fileItem.url = data
}
```
想知道组件就是这样设计的，需要在 `success` 手动更新一次真实 `url`, 还是说是 `bug` 呢。

ps: 看了下源码，如果不自定义上传， 也没看到上传成功后会更新对应 `fileItem` 的 `url`。

**Comment by eiinu** [comment]
Uploader 组件并不能更新这个数据，因为每个开发者服务器的返回值（例如自定义方法 uploadFileToServer 的返回值）的数据结构可能是不同的。

**Reply by GreatAuk** [reply]
> Uploader 组件并不能更新这个数据，因为每个开发者服务器的返回值（例如自定义方法 uploadFileToServer 的返回值）的数据结构可能是不同的。

那我的做法没错了？那最好在文档对应的 demo 写明来...。
话说如果不自定义上传，直接设置上传的 url, 也不会更新吗。

**Reply by eiinu** [reply, marked as answer]
> > Uploader 组件并不能更新这个数据，因为每个开发者服务器的返回值（例如自定义方法 uploadFileToServer 的返回值）的数据结构可能是不同的。
> 
> 那我的做法没错了？那最好在文档对应的 demo 写明来...。 话说如果不自定义上传，直接设置上传的 url, 也不会更新吗。

我的意思是，每个开发者编写的服务器上传API都是不同的，你这里会返回上传后的文件链接，其他人可能会返回其他的数据结构，甚至有可能不返回任何数据。组件本身是没有能力处理这种情况的，唯一能做的只是把接口返回值抛出，由开发者自行处理后续的逻辑。
